sdks/python/apache_beam/runners/interactive/background_caching_job.py

```diff
@@ -19,29 +19,113 @@
 
 For internal use only; no backwards-compatibility guarantees.
 
-A background caching job is a job that caches events for all unbounded sources
-of a given pipeline. With Interactive Beam, one such job is started when a
-pipeline run happens (which produces a main job in contrast to the background
+A background caching job is a job that captures events for all capturable
+sources of a given pipeline. With Interactive Beam, one such job is started when
+a pipeline run happens (which produces a main job in contrast to the background
 caching job) and meets the following conditions:
 
-  #. The pipeline contains unbounded sources.
+  #. The pipeline contains capturable sources, configured through
+     interactive_beam.options.capturable_sources.
   #. No such background job is running.
   #. No such background job has completed successfully and the cached events are
-     still valid (invalidated when unbounded sources change in the pipeline).
+     still valid (invalidated when capturable sources change in the pipeline).
 
 Once started, the background caching job runs asynchronously until it hits some
-cache size limit. Meanwhile, the main job and future main jobs from the pipeline
-will run using the deterministic replay-able cached events until they are
-invalidated.
+capture limit configured in interactive_beam.options. Meanwhile, the main job
+and future main jobs from the pipeline will run using the deterministic
+replayable captured events until they are invalidated.
 """
 
 # pytype: skip-file
 
 from __future__ import absolute_import
 
+import logging
+import threading
+import time
+
 import apache_beam as beam
-from apache_beam import runners
 from apache_beam.runners.interactive import interactive_environment as ie
+from apache_beam.runners.interactive.caching import streaming_cache
+from apache_beam.runners.runner import PipelineState
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
+
+
+class BackgroundCachingJob(object):
+  """A simple abstraction that controls necessary components of a timed and
+  [disk] space limited background caching job.
```

Comment:
This is to indicate the captured data could be on-disk, or in other mediums such as in-memory (some testing cache manager implementation).
Let me just remove the `[disk]` to avoid the confusion.

Reply:
OK